### PR TITLE
REPO-4648: Update doc Dockerfile when creating image upon repository content image

### PR DESCRIPTION
**Pull request by oussamaMesseguem**
Shows how we do proceed if we'd like to add an amp to u custom image
based on repository content image
Switching user from alfresco to root to alfresco is required

### CODE REVIEW

File: docs/create-custom-image-using-existing-docker-image.md [hunk 1]
```diff
@@ -40,29 +40,53 @@ mvn clean install
...
 ```bash
-### Apply AMPs to the latest Community repository image.
-FROM alfresco/alfresco-content-repository-community:latest
+# Apply AMPs to the latest Community repository image.
+ARG BASE_IMAGE=alfresco/alfresco-content-repository-community
+ARG VERSION=latest
+
+FROM ${BASE_IMAGE}:${VERSION}
+
+# Need these viariables to give permissions to tomcat (refer Dockerfile below)
```
Comment (killerboot) [inline docs/create-custom-image-using-existing-docker-image.md]: Please correct to:
"Default user and group are used to setup permissions for Tomcat process, see parent Dockerfile: [Alfresco/acs-community-packaging/docker-alfresco/Dockerfile](https://github.com/Alfresco/acs-community-packaging/blob/master/docker-alfresco/Dockerfile#L5)

File: docs/create-custom-image-using-existing-docker-image.md [hunk 2]
```diff
@@ -40,29 +40,53 @@ mvn clean install
...
-COPY amps /usr/local/tomcat/amps
+COPY amps ${TOMCAT_DIR}/amps
 
 ### Install AMPs on alfresco
-RUN java -jar /usr/local/tomcat/alfresco-mmt/alfresco-mmt*.jar install \
-              /usr/local/tomcat/amps/alfresco-bulk-import-*.amp /usr/local/tomcat/webapps/alfresco -nobackup -force
+RUN java -jar ${TOMCAT_DIR}/alfresco-mmt/alfresco-mmt*.jar install \
+              ${TOMCAT_DIR}/amps ${TOMCAT_DIR}/webapps/alfresco -directory -nobackup -force
+
+# Need to have all Tomcat files owned by root with group GROUPNAME as mentionned (refer Dockerfile below)
```
Comment (killerboot) [inline docs/create-custom-image-using-existing-docker-image.md]: Please correct to:
"All files in the tomcat folder must be owned by root user and Alfresco group as mentioned in the parent Dockerfile"

File: docs/create-custom-image-using-existing-docker-image.md [hunk 3]
```diff
@@ -40,29 +40,53 @@ mvn clean install
...
+    find ${TOMCAT_DIR}/webapps -type f -exec chmod 0640 {} \; && \
+    chmod -R g+r ${TOMCAT_DIR}/webapps && \
+    chgrp -R ${GROUPNAME} ${TOMCAT_DIR}
+
+# Switching back to alfresco user after having added amps files to run the container as non-root
+USER ${USERNAME}
+
 ```
 
+** Note: ** [Alfresco/acs-community-packaging/docker-alfresco/Dockerfile](https://github.com/Alfresco/acs-community-packaging/blob/master/docker-alfresco/Dockerfile).
```
Comment (killerboot) [inline docs/create-custom-image-using-existing-docker-image.md]: Please clarify what this link is for:

** Note ** Parent Dockerfile is located here: [Alfresco/acs-community-packaging/docker-alfresco/Dockerfile](https://github.com/Alfresco/acs-community-packaging/blob/master/docker-alfresco/Dockerfile)

File: docs/create-custom-image-using-existing-docker-image.md [hunk 4]
```diff
@@ -40,29 +40,53 @@ mvn clean install
...
+
 ** Note: ** It's not necessary to build the .amp file if you can download it via a network (FTP/HTTP/HTTPS), since the COPY command can be replaced by CURL/WGET.
 
 * Open a Terminal and run the following command to build the custom docker image.
 
 ```bash
-docker build . -t customrepository/alfresco-custom-image:customTag
+docker build -t <repository>:<tag> .
```
Comment (killerboot) [inline docs/create-custom-image-using-existing-docker-image.md]: Why is this required to be changed?

File: docs/create-custom-image-using-existing-docker-image.md [hunk 5]
```diff
@@ -115,19 +139,29 @@ alfresco-custom-image                            (Dir)
 * Create a Dockerfile for custom-alfresco-repository image:
 
 ```bash
-### Apply AMPs to the latest Enterprise repository image.
-FROM alfresco/alfresco-content-repository:latest
```
Comment (killerboot) [inline docs/create-custom-image-using-existing-docker-image.md]: This change is not required.

File: docs/create-custom-image-using-existing-docker-image.md [hunk 6]
```diff
@@ -150,19 +184,35 @@ RUN java -jar /usr/local/tomcat/alfresco-mmt/alfresco-mmt*.jar install \
...
+RUN sed -i "s/\    <\/Engine>/\n\    <\/Engine>\n\    <Connector\ port=\"8443\"\n\               protocol=\"org.apache.coyote.http11.Http11Nio2Protocol\"\n\               sslImplementationName=\"org.apache.tomcat.util.net.jsse.JSSEImplementation\"\n\               maxThreads=\"150\"\n\               SSLEnabled=\"true\">\n\        <SSLHostConfig certificateVerification=\"required\" \n\                       truststoreFile=\"\/usr\/local\/tomcat\/keystore\/ssl.keystore\/ssl.truststore\"\n\                       truststorePassword=\"truststorePass\"\n\                       truststoreType=\"JCEKS\" >\n\            <Certificate certificateKeystoreFile=\"\/usr\/local\/tomcat\/keystore\/ssl.keystore\/ssl.keystore\"\n\                         certificateKeystorePassword=\"keystorePass\"\n\                         certificateKeystoreType=\"JCEKS\" \/>\n\        <\/SSLHostConfig>\n\    <\/Connector>/g" ${TOMCAT_DIR}/conf/server.xml
+
+
+# Need to have all Tomcat files owned by root with group GROUPNAME as mentionned (refer Dockerfile below)
```
Comment (killerboot) [inline docs/create-custom-image-using-existing-docker-image.md]: Please correct to:
"All files in the tomcat folder must be owned by root user and Alfresco group as mentioned in the parent Dockerfile"

File: docs/create-custom-image-using-existing-docker-image.md [hunk 7]
```diff
@@ -150,19 +184,35 @@ RUN java -jar /usr/local/tomcat/alfresco-mmt/alfresco-mmt*.jar install \
...
+
+# Switching back to alfresco user after having added amps files to run the container as non-root
+USER ${USERNAME}
+
 ```
 
 * Open a Terminal and run the following command to build the custom docker repository image.
 
 ```bash
-docker build . -t customrepository/custom-alfresco-repository:customTag
```
Comment (killerboot) [inline docs/create-custom-image-using-existing-docker-image.md]: This change is not required.

File: docs/create-custom-image-using-existing-docker-image.md [hunk 8]
```diff
@@ -171,18 +221,26 @@ docker push customrepository/custom-alfresco-repository:customTag
 ### Apply AMPs to the latest Share image (at this moment).
 FROM alfresco/alfresco-share:6.0.0-rc2
 
-### Copy the keystore build context to the appropriate location for your application server
-COPY keystore /usr/local/tomcat/keystore
+# Need these viariables to give permissions to tomcat (refer Dockerfile below)
```
Comment (killerboot) [inline docs/create-custom-image-using-existing-docker-image.md]: Please correct to:
"Default user and group are used to setup permissions for Tomcat process, see parent Dockerfile: Alfresco/acs-community-packaging/docker-alfresco/Dockerfile

File: docs/create-custom-image-using-existing-docker-image.md [hunk 9]
```diff
@@ -171,18 +221,26 @@ docker push customrepository/custom-alfresco-repository:customTag
...
+ARG GROUPNAME=Alfresco
+ARG USERNAME=alfresco
+ARG TOMCAT_DIR=/usr/local/tomcat
+
+# Need to switch to root to add the new files
```
Comment (killerboot) [inline docs/create-custom-image-using-existing-docker-image.md]: Please correct to:

Alfresco user does not have permissions to modify webapps or configuration. Switch to root.
The access will be fixed after all operations are done.

File: docs/create-custom-image-using-existing-docker-image.md [hunk 10]
```diff
@@ -115,19 +139,29 @@ alfresco-custom-image                            (Dir)
...
+ARG VERSION=latest
+
+FROM ${BASE_IMAGE}:${VERSION}
+
+# Need these viariables to give permissions to tomcat (refer Dockerfile below)
+ARG GROUPNAME=Alfresco
+ARG USERNAME=alfresco
+ARG TOMCAT_DIR=/usr/local/tomcat
+
+# Need to switch to root to add the new files
```
Comment (killerboot) [inline docs/create-custom-image-using-existing-docker-image.md]: Please correct to:

Alfresco user does not have permissions to modify webapps or configuration. Switch to root.
The access will be fixed after all operations are done.

File: docs/create-custom-image-using-existing-docker-image.md [hunk 11]
```diff
@@ -40,29 +40,53 @@ mvn clean install
...
```
Comment (killerboot) [inline docs/create-custom-image-using-existing-docker-image.md]: Please correct to:

Alfresco user does not have permissions to modify webapps or configuration. Switch to root.
The access will be fixed after all operations are done.

File: docs/create-custom-image-using-existing-docker-image.md [hunk 12]
```diff
@@ -205,17 +263,34 @@ RUN java -jar /usr/local/tomcat/alfresco-mmt/alfresco-mmt*.jar install \
...
+RUN sed -i "s/\    <\/Engine>/\n\    <\/Engine>\n\    <Connector\ port=\"8443\"\n\               protocol=\"org.apache.coyote.http11.Http11Nio2Protocol\"\n\               sslImplementationName=\"org.apache.tomcat.util.net.jsse.JSSEImplementation\"\n\               maxThreads=\"150\"\n\               SSLEnabled=\"true\">\n\        <SSLHostConfig certificateVerification=\"required\" \n\                       truststoreFile=\"\/usr\/local\/tomcat\/keystore\/ssl.keystore\/ssl.truststore\"\n\                       truststorePassword=\"truststorePass\"\n\                       truststoreType=\"JCEKS\" >\n\            <Certificate certificateKeystoreFile=\"\/usr\/local\/tomcat\/keystore\/ssl.keystore\/ssl.keystore\"\n\                         certificateKeystorePassword=\"keystorePass\"\n\                         certificateKeystoreType=\"JCEKS\" \/>\n\        <\/SSLHostConfig>\n\    <\/Connector>/g" ${TOMCAT_DIR}/conf/server.xml
+
+
+# Need to have all Tomcat files owned by root with group GROUPNAME as mentionned (refer Dockerfile below)
```
Comment (killerboot) [inline docs/create-custom-image-using-existing-docker-image.md]: Please correct to:
"All files in the tomcat folder must be owned by root user and Alfresco group as mentioned in the parent Dockerfile"

File: docs/create-custom-image-using-existing-docker-image.md [hunk 13]
```diff
@@ -40,29 +40,53 @@ mvn clean install
 * Add the following lines to the Dockerfile:
 
 ```bash
-### Apply AMPs to the latest Community repository image.
-FROM alfresco/alfresco-content-repository-community:latest
+# Apply AMPs to the latest Community repository image.
+ARG BASE_IMAGE=alfresco/alfresco-content-repository-community
+ARG VERSION=latest
+
+FROM ${BASE_IMAGE}:${VERSION}
```
[the remark this answers is not in the document]
Comment (oussamaMesseguem) [reply docs/create-custom-image-using-existing-docker-image.md]: Okay

File: docs/create-custom-image-using-existing-docker-image.md (hunk 3 above)
Comment (oussamaMesseguem) [reply docs/create-custom-image-using-existing-docker-image.md]: Can we put a link in the bash spot code?

File: docs/create-custom-image-using-existing-docker-image.md (hunk 4 above)
Comment (oussamaMesseguem) [reply docs/create-custom-image-using-existing-docker-image.md]: It's the way we use for abstract names instead of giving a random example.
It's not required but more standard.
